src/modules/checkIsValidUserInput.js

```diff
@@ -8,8 +8,48 @@
  * @param {string} userInput - The user input
  * @return {boolean} - True if the user input is valid, false otherwise
  */
+const checkIsUniqueNumbers = (value) => {
+  const arrayValue = `${value}`.split('');
+
+  if (arrayValue.length !== 4 && typeof value !== 'number') {
+    return false;
+  }
+
+  if (arrayValue[0] === '0') {
+    return false;
+  }
+
+  for (let i = 0; i < arrayValue.length; i++) {
+    const numberToCheck = arrayValue[i];
+
+    arrayValue.splice(i, 1);
+
+    if (arrayValue.includes(numberToCheck)) {
+      return false;
+    }
+
+    return true;
+  }
```

Comment:
The loop in `checkIsUniqueNumbers` returns true prematurely after the first iteration. It should only return true after checking all digits. Also, modifying the array with `splice` during iteration can lead to incorrect results. Consider using a different approach to check for duplicates without altering the array.